Reorganising the changelog for clarity plus minor edits (#7638)
tinymce/tinymce: modules/tinymce/CHANGELOG.md

### Added
- New `editor.options` API to replace the old `editor.settings` and `editor.getParam` APIs #TINY-8206
- New `editor.annotator.removeAll` API to remove all annotations by name #TINY-8195
- New `Resource.unload` API to make it possible to unload resources #TINY-8431
- New `FakeClipboard` API on the `tinymce` global #TINY-8353

> - New `dispatch()` function to replace the now deprecated `fire()` function in various APIs #TINY-8102

Reviewer: Should this be "change" instead of New? dispatch() replaces fire()

Reviewer 2: We didn't remove the `fire` function so the `dispatch` function is actually new.

Author: Hmm, true. I think it was basically a rename? Not entirely sure

Reviewer 2: It sort of was but largely wasn't, as the change added a new function and then replaced all internal uses to use the new name, instead of the deprecated one. Then as a last step it deprecated the old function name, which is why there's an entry in Deprecated for that as well.

Author: Right, makes sense. We should probably keep it as it is then

- Removed the `imagetools` dialog component #TINY-8333
- Removed the `toc` plugin, which is now classified as a Premium plugin #TINY-8250
- Removed the `tabfocus` plugin #TINY-8315
- Removed the `textpattern` plugin's API as part of moving it to core #TINY-8312
- Removed the `table` plugin's API #TINY-8273
- Removed the callback for the `EditorUpload` API #TINY-8325
- Removed the `filterNode` method from the `DomParser` API #TINY-8249
- Removed the `SaxParser` API #TINY-8218
- Removed the `tinymce.utils.Promise` API #TINY-8241
- Removed the `toHex` function for the `DOMUtils` and `Styles` APIs #TINY-8163
- Removed the `execCommand` handler function from the plugin and theme interfaces #TINY-7829
- Removed the `editor.settings` property as it has been replaced by the new Options API #TINY-8236
- Removed the `shortEnded` and `fixed` properties on `tinymce.html.Node` class #TINY-8205
- Removed the `mceInsertRawHTML` command #TINY-8214
- Removed the style field from the `image` plugin dialog advanced tab #TINY-3422
- Removed the legacy `mobile` theme #TINY-7832
- Removed the legacy browser detection properties from `Env` #TINY-8162

> - Removed support for Microsoft Internet Explorer 11 #TINY-8194 #TINY-8241

Reviewer 2: Should this be up the top since it's one of the key things we need to highlight?

Author: I'd be fine with that. Can do

Author: Done

- New `slider` dialog component #TINY-8304
- New `imagepreview` dialog component, allowing preview and zoom of any image URL #TINY-8333
- New `buttonType` property on dialog button components, supporting `toolbar` style in addition to `primary` and `secondary` #TINY-8304
- The `tabindex` attribute is now copied from the target element to the iframe #TINY-8315

### Improved
- The default height of editor has been increased from `200px` to `400px` to improve the usability of the editor #TINY-6860
- The upload results returned from the `editor.uploadImages()` API now includes a `removed` flag, reflecting if the image was removed after a failed upload #TINY-7735
- The `ScriptLoader`, `StyleSheetLoader`, `AddOnManager`, `PluginManager` and `ThemeManager` APIs will now return a `Promise` when loading resources instead of using callbacks #TINY-8325
- A `ThemeLoadError` event is now fired if the theme fails to load #TINY-8325
- The `BeforeSetContent` event will now include the actual serialized content when passing in an `AstNode` to the `editor.setContent` API #TINY-7996
- Improved support for placing the caret before or after noneditable elements within the editor #TINY-8169

> - Calls to `editor.selection.setRng` now update the cursor position bookmark used when focus is returned to the editor #TINY-8450

Reviewer 2: ```suggestion
- Calls to `editor.selection.setRng` now update the caret position bookmark used when focus is returned to the editor #TINY-8450
```
Just noticed this uses the wrong term, as cursor is the mouse pointer.

- The `images_upload_handler` option is no longer passed a `success` or `failure` callback and instead requires a `Promise` to be returned with the upload result #TINY-8325
- The `tinymce.settings` global property is no longer set upon initialization #TINY-7359
- The `change` event is no longer fired on first modification #TINY-6920
- The `GetContent` event will now always pass a `string` for the `content` property #TINY-7996
- Changed the default tag for the strikethrough format to the `s` tag when using a html 5 schema #TINY-8262
- The `strike` tag is automatically converted to the `s` tag when using a html 5 schema #TINY-8262
- Aligning a table to the left or right will now use margin styling instead of float styling #TINY-6558
- The `:` control character has been changed to `~` for the schema `valid_elements` and `extended_valid_elements` options #TINY-6726
- The `primary` property on dialog buttons has been deprecated. Use the new `buttonType` property instead #TINY-8304
- Changed the default statusbar element path delimiter from `»` to `›` #TINY-8372

> - Replaced 'Powered by Tiny' link text with logo #TINY-8371

Reviewer 2: ```suggestion
- Replaced the `Powered by Tiny` link text with the Tiny logo #TINY-8371
```
Backticks instead of quotes for this maybe? Also I don't think this fully made sense, so made a minor change there.

- The `editor.selection.getContent()` API did not respect the `no_events` argument #TINY-8018
- The `editor.annotator.remove` API did not keep selection when removing the annotation #TINY-8195
- The `GetContent` event was not fired when getting `tree` or `text` formats using the `editor.selection.getContent()` API #TINY-8018
- The `beforeinput` and `input` events would sometimes not fire as expected when deleting content #TINY-8168 #TINY-8329
- The `table` plugin would sometimes not correctly handle headers in the `tfoot` section #TINY-8104
- The `silver` theme UI was incorrectly rendered before plugins had initialized #TINY-8288
- The aria labels for the color picker dialog were not translated #TINY-8381
- Fixed sub-menu items not read by screen readers. Patch contributed by westonkd #TINY-8417
- Dialog labels and other text-based UI properties did not escape HTML markup #TINY-7524
- Anchor elements would render incorrectly when using the `allow_html_in_named_anchor` option #TINY-3799
- The Home or End keys would move out of a editable element contained within a noneditable element #TINY-8201

### Removed
- Removed support for Microsoft Internet Explorer 11 #TINY-8194 #TINY-8241

> - Removed support for Word from the `paste` plugin #TINY-7493

Reviewer 2: Does this need rewording now that we no longer have a paste plugin?

Author: I honestly can not think of a good way to reword this to clarify that

Reviewer 2: Yeah nor can I to be honest. The best I could come up with was something like this... Let's leave it as is then if you don't like the suggestion or would prefer to leave it as is.
```suggestion
- Removed support for Word from the opensource paste functionality #TINY-7493
```

Reviewer 3: I suggest `Microsoft Word` instead of just `Word`

Reviewer 2: That's a good point, yeah it should definitely be the full product name. I'll update my suggestion to include that.

Author: I'll go with this, thanks!
